test/ecto/adapters/sqlite3/codec_test.exs

```diff
@@ -176,4 +176,21 @@ defmodule Ecto.Adapters.SQLite3.CodecTest do
       end
     end
   end
+
+  describe ".date_encode/2" do
+    setup do
+      [
+        date: ~D[2011-01-09],
+        datetime: ~U[2011-01-09 08:46:08.00Z]
+      ]
+    end
+
+    test "on %Date{} structs", %{date: date} do
+      {:ok, "2011-01-09"} = Codec.date_encode(date, :iso8601)
+    end
+
+    test "on %DateTime{} structs", %{datetime: datetime} do
+      {:ok, "2011-01-09"} = Codec.date_encode(datetime, :iso8601)
+    end
```

Comment:
I'm not sure if date_encode() should support `%DateTime{}` structs, though.

Reply:
Should go check what the postgres adapter does to support this. I think we can truncate and be fine, but that seems like it may cause confusion down the road rather than blow up.

Reply:
Let's keep it specific to Date structs then. We can expand it in the future if there's a real use case to accept DateTime structs as values in :date fields.